### ISSUE TITLE
Spring Boot 3.2 + Thymeleaf: EL1005E Type cannot be found for valid classes

### ISSUE THREAD

**Issue by gboersma** [spring-projects/spring-boot#38574]
I have a Spring Boot application that runs as expected with Spring Boot 3.1.6. Upgrading to 3.2.0 now causes Thymeleaf exceptions on any page that uses the "T(" + class name (EL1005E: Type cannot be found).

Any ideas?

This is in my pom.xml:
```
        <dependency>
            <groupId>org.springframework.boot</groupId>
            <artifactId>spring-boot-starter-thymeleaf</artifactId>
        </dependency>
        <dependency>
            <groupId>nz.net.ultraq.thymeleaf</groupId>
            <artifactId>thymeleaf-layout-dialect</artifactId>
        </dependency>
        <dependency>
            <groupId>org.thymeleaf.extras</groupId>
            <artifactId>thymeleaf-extras-springsecurity6</artifactId>
        </dependency>
```

On my template, I have the following:
```
<span th:if="${T(com.icsynergy.icrare2.domain.search.SearchAttribute).isUserUtilMethodAttr(searchResultAttr.field)}" th:utext="${userUtilClass.__${searchAttribute.formatUserUtilMethodAttr(searchResultAttr.field)}__(user,searchModel)}">{{attr}}</span>
```

I get the error:
```
org.thymeleaf.exceptions.TemplateInputException: An error happened during template parsing (template: "class path resource [templates/search/_user_results.html]")
	at org.thymeleaf.templateparser.markup.AbstractMarkupTemplateParser.parse(AbstractMarkupTemplateParser.java:241) ~[thymeleaf-3.1.2.RELEASE.jar:3.1.2.RELEASE]
	at org.thymeleaf.templateparser.markup.AbstractMarkupTemplateParser.parseStandalone(AbstractMarkupTemplateParser.java:100) ~[thymeleaf-3.1.2.RELEASE.jar:3.1.2.RELEASE]
	at org.thymeleaf.engine.TemplateManager.parseAndProcess(TemplateManager.java:666) ~[thymeleaf-3.1.2.RELEASE.jar:3.1.2.RELEASE]
	at org.thymeleaf.TemplateEngine.process(TemplateEngine.java:1103) ~[thymeleaf-3.1.2.RELEASE.jar:3.1.2.RELEASE]
	at org.thymeleaf.TemplateEngine.process(TemplateEngine.java:1077) ~[thymeleaf-3.1.2.RELEASE.jar:3.1.2.RELEASE]
	at org.thymeleaf.spring6.view.ThymeleafView.renderFragment(ThymeleafView.java:372) ~[thymeleaf-spring6-3.1.2.RELEASE.jar:3.1.2.RELEASE]
	at org.thymeleaf.spring6.view.ThymeleafView.render(ThymeleafView.java:192) ~[thymeleaf-spring6-3.1.2.RELEASE.jar:3.1.2.RELEASE]
	at org.springframework.web.servlet.DispatcherServlet.render(DispatcherServlet.java:1431) ~[spring-webmvc-6.1.1.jar:6.1.1]
	at org.springframework.web.servlet.DispatcherServlet.processDispatchResult(DispatcherServlet.java:1167) ~[spring-webmvc-6.1.1.jar:6.1.1]
	at org.springframework.web.servlet.DispatcherServlet.doDispatch(DispatcherServlet.java:1106) ~[spring-webmvc-6.1.1.jar:6.1.1]
	at org.springframework.web.servlet.DispatcherServlet.doService(DispatcherServlet.java:979) ~[spring-webmvc-6.1.1.jar:6.1.1]
	at org.springframework.web.servlet.FrameworkServlet.processRequest(FrameworkServlet.java:1014) ~[spring-webmvc-6.1.1.jar:6.1.1]
	at org.springframework.web.servlet.FrameworkServlet.doPost(FrameworkServlet.java:914) ~[spring-webmvc-6.1.1.jar:6.1.1]
...
Caused by: org.thymeleaf.exceptions.TemplateProcessingException: Exception evaluating SpringEL expression: "T(com.icsynergy.icrare2.domain.search.SearchAttribute).isUserUtilMethodAttr(searchResultAttr.field)" (template: "search/_user_results" - line 91, col 39)
	at org.thymeleaf.spring6.expression.SPELVariableExpressionEvaluator.evaluate(SPELVariableExpressionEvaluator.java:292) ~[thymeleaf-spring6-3.1.2.RELEASE.jar:3.1.2.RELEASE]
	at org.thymeleaf.standard.expression.VariableExpression.executeVariableExpression(VariableExpression.java:166) ~[thymeleaf-3.1.2.RELEASE.jar:3.1.2.RELEASE]
	at org.thymeleaf.standard.expression.SimpleExpression.executeSimple(SimpleExpression.java:66) ~[thymeleaf-3.1.2.RELEASE.jar:3.1.2.RELEASE]
	at org.thymeleaf.standard.expression.Expression.execute(Expression.java:109) ~[thymeleaf-3.1.2.RELEASE.jar:3.1.2.RELEASE]
...
Caused by: org.springframework.expression.spel.SpelEvaluationException: EL1005E: Type cannot be found 'com.icsynergy.icrare2.domain.search.SearchAttribute'
	at org.springframework.expression.spel.support.StandardTypeLocator.findType(StandardTypeLocator.java:128)
```

**Comment by gboersma** [comment]
Thanks! I tried with Spring Framework 6.1.2-SNAPSHOT and all works as expected.

**Comment by gboersma** [comment]
Duplicate of https://github.com/spring-projects/spring-framework/issues/31668

**Comment by gboersma** [comment]
From my perspective, it's closed because I have a workaround. However, anyone that grabs Spring Boot 3.2 will have the same issue. So perhaps you want to resolve a different way.

Please close as you see fit.

**Comment by gboersma** [comment]
I also confirmed that this one is fixed in Spring Boot 3.2.1-SNAPSHOT.

**Comment by wilkinsona** [comment]
Thanks, @gboersma. Much appreciated.
